src/main/java/com/asu/cse545/group12/controller/UserController.java

```diff
@@ -63,13 +77,30 @@ public ModelAndView getForm() {
 		}
 		ModelAndView modelView = new ModelAndView();
 		modelView.addObject("user", new Users());
-		modelView.addObject("userPII", new UserPII());
+		modelView.addObject("userpii", new UserPII());
 		modelView.setViewName("signup");
 		
 		return modelView;
 	}
 
-	@RequestMapping("signUpExternalUser")
+	@RequestMapping( value= "signUpExternalUser")
+
+	public ModelAndView registerUser(@ModelAttribute Users user) {
+		//Users user=(Users)modelMap.get("user");
+		user.setUserStatus("A");
+		java.util.Date date = Calendar.getInstance().getTime();
+		user.setLastModifieddate(date);
+		user.setRegistrationDate(date);
+		System.out.println("9999"+user.getUserpii());
+		user.setUserId(user.getUserpii().getUserId());
+		user.getUserpii().setDateOfBirth(date);
```

Comment:
This logic of setting user fields should actually be done in service layer. controller should not deal with this. instead controller will do validation stuff and once valid it should call service layer to do this logic